### PR TITLE
User Management

### PR DESCRIPTION
**Pull request by rn0**

### CODE REVIEW

File: Admin/PasswordResetBatchElement.php [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+    public function apply($object)
+    {
+        if (!$object instanceof ResettablePasswordInterface) {
+            throw new \InvalidArgumentException(
+                'Object should implement FSi\Bundle\AdminSecurityBundle\Security\User\ResettablePasswordInterface'
+            );
+        }
+
+        $object->setPasswordResetToken($this->tokenFactory->createToken());
+        $this->mailer->send($object);
```
Comment (rn0) [inline Admin/PasswordResetBatchElement.php]: We should dispatch `AdminSecurityEvents::RESET_PASSWORD_REQUEST` event

File: Admin/UserElement.php [hunk 2]
```diff
@@ -0,0 +1,64 @@
+<?php
+
+namespace FSi\Bundle\AdminSecurityBundle\Admin;
```
Comment (rn0) [inline Admin/UserElement.php]: This should be in `Doctrine\Admin` namespace

File: EventListener/PrepareUserListener.php [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public static function getSubscribedEvents()
+    {
+        return array(
+            FormEvents::FORM_DATA_PRE_SAVE => array(
+                array('setEmailAsUsernameAndEnableUser', 1),
```
Comment (rn0) [inline EventListener/PrepareUserListener.php]: This should be done in subscriber for AdminSecurityEvents::USER_CREATED

File: EventListener/PrepareUserListener.php [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+                array('setRandomPassword', 2),
```
Comment (rn0) [inline EventListener/PrepareUserListener.php]: `FormEvents::FORM_DATA_PRE_SAVE` fires on create & update. `setRandomPassword` should be invoked only on create

**Comment by rn0** [comment]
@fsi-open

File: EventListener/PrepareUserListener.php [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+    public static function getSubscribedEvents()
+    {
+        return array(
+            FormEvents::FORM_DATA_PRE_SAVE => array(
+                'prepareAndDispatchUserCreated',
+            )
+        );
+    }
+
+    public function prepareAndDispatchUserCreated(FormEvent $event)
```
Comment (rn0) [inline EventListener/PrepareUserListener.php]: this method should detect if entity is new

File: Resources/config/orm.xml [hunk 6]
```diff
@@ -5,7 +5,7 @@
            xsi:schemaLocation="http://symfony.com/schema/dic/services http://symfony.com/schema/dic/services/services-1.0.xsd">
 
     <parameters>
-        <parameter key="admin_security.repository.user.class">FSi\Bundle\AdminSecurityBundle\Doctrine\UserRepository</parameter>
+        <parameter key="admin_security.repository.user.class">FSi\Bundle\AdminSecurityBundle\Doctrine\RepositoryUserRepository</parameter>
```
Comment (rn0) [inline Resources/config/orm.xml]: Missing `\` -> `Repository\UserRepository`

**Comment by rn0** [comment]
@fsi-open

File: Form/Type/Admin/UserType.php [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+
+use Symfony\Component\Form\AbstractType;
+use Symfony\Component\Form\FormBuilderInterface;
+use Symfony\Component\OptionsResolver\OptionsResolver;
+use Symfony\Component\OptionsResolver\OptionsResolverInterface;
+
+class UserType extends AbstractType
+{
+    /**
+     * @var
```
Comment (szymach) [inline Form/Type/Admin/UserType.php]: missing value for @var

**Comment by rn0** [comment]
@fsi-open

**Comment by szymach** [comment]
:+1:

**Comment by rn0** [comment]
@fsi-open

File: EventListener/SetEmailAsUsernameListener.php [hunk 8]
```diff
@@ -0,0 +1,43 @@
+<?php
+/**
+ * Created by PhpStorm.
+ * User: piotr
+ * Date: 8/27/15
+ * Time: 9:49 AM
+ */
```
Comment (chives) [inline EventListener/SetEmailAsUsernameListener.php]: please change this docblock to standard FSi docblock

**Comment by chives** [comment]
:+1:

**Comment by JarekW** [comment]
:+1:

**Pull request merged**